### PR TITLE
Vehicle routing quickstart: rename Customer to Visit, remove Depot

### PR DESCRIPTION
**Pull request by pieterdeschepper**
## Description of the change

Remove Depots in favour of homeLocation on the Vehicle Object and generalize Customers to Visits, as requested in https://github.com/TimefoldAI/timefold-quickstarts/issues/237

### Development

- [x] The changes have been covered with tests, if necessary.
- [x] You have a green build, with the exception of the flaky tests.
- [x] UI and JS files are fully tested, the user interface works for all modules affected by your changes (e.g., solve and analyze buttons).
- [x] The network calls work for all modules affected by your changes (e.g., solving a problem).
- [x] The console messages are validated for all modules affected by your changes.

### Code Review

- [x] This pull request includes an explanatory title and description.
- [x] The GitHub issue is linked.
- [ ] At least one other engineer has approved the changes.
- [ ] After PR is merged, inform the reporter.

### CODE REVIEW

**Comment by triceo** [comment]
@zepfred Please review at your earliest convenience.

File: use-cases/vehicle-routing/src/main/java/org/acme/vehiclerouting/rest/VehicleRouteDemoResource.java [hunk 1]
```diff
@@ -45,22 +42,21 @@ public class VehicleRouteDemoResource {
     private static final LocalTime AFTERNOON_WINDOW_END = LocalTime.of(18, 0);
 
     public enum DemoData {
-        PHILADELPHIA(0, 60, 6, 2, LocalTime.of(7, 30),
+        PHILADELPHIA(0, 55, 6, LocalTime.of(7, 30),
```
Comment (zepfred) [inline use-cases/vehicle-routing/src/main/java/org/acme/vehiclerouting/rest/VehicleRouteDemoResource.java:47]: Was the change to 55 made intentionally?

Comment (pieterdeschepper) [reply use-cases/vehicle-routing/src/main/java/org/acme/vehiclerouting/rest/VehicleRouteDemoResource.java:47]: Yes, given that there are now more homeLocations than there were depots, the solver failed to find a feasible solution with 60 visits. Reducing it to 55 visits makes sure that there is a feasible solution + gives some bandwidth for planning recommended fit visits

File: use-cases/vehicle-routing/src/main/java/org/acme/vehiclerouting/rest/VehicleRoutePlanResource.java [hunk 2]
```diff
@@ -108,34 +108,34 @@ public String solve(VehicleRoutePlan problem) {
...
-                .filter(c -> c.getId().equals(request.customerId()))
+        Visit visit = request.solution().getVisits().stream()
+                .filter(c -> c.getId().equals(request.visitId()))
                 .findFirst()
-                .orElseThrow(() -> new IllegalStateException("Customer %s not found".formatted(request.customerId())));
+                .orElseThrow(() -> new IllegalStateException("Visit %s not found".formatted(request.visitId())));
         List<RecommendedFit<VehicleRecommendation, HardSoftLongScore>> recommendedFitList = solutionManager
-                .recommendFit(request.solution(), customer, c -> new VehicleRecommendation(c.getVehicle().getId(),
-                        c.getVehicle().getCustomers().indexOf(c)));
+                .recommendFit(request.solution(), visit, c -> new VehicleRecommendation(c.getVehicle().getId(),
```
Comment (zepfred) [inline use-cases/vehicle-routing/src/main/java/org/acme/vehiclerouting/rest/VehicleRoutePlanResource.java]:
```suggestion
                .recommendFit(request.solution(), visit, v -> new VehicleRecommendation(v.getVehicle().getId(),
```

File: use-cases/vehicle-routing/src/main/java/org/acme/vehiclerouting/rest/VehicleRoutePlanResource.java [hunk 3]
```diff
@@ -108,34 +108,34 @@ public String solve(VehicleRoutePlan problem) {
...
+                        c.getVehicle().getVisits().indexOf(c)));
```
Comment (zepfred) [inline use-cases/vehicle-routing/src/main/java/org/acme/vehiclerouting/rest/VehicleRoutePlanResource.java]:
```suggestion
                        v.getVehicle().getVisits().indexOf(v)));
```

File: use-cases/vehicle-routing/src/main/java/org/acme/vehiclerouting/rest/VehicleRoutePlanResource.java [hunk 4]
```diff
@@ -149,11 +149,11 @@ public VehicleRoutePlan applyRecommendedFit(ApplyRecommendationRequest request)
                 .filter(v -> v.getId().equals(vehicleId))
                 .findFirst()
                 .orElseThrow(() -> new IllegalStateException("Vehicle %s not found".formatted(vehicleId)));
-        Customer customer = request.solution().getCustomers().stream()
-                .filter(c -> c.getId().equals(request.customerId()))
+        Visit visit = request.solution().getVisits().stream()
+                .filter(c -> c.getId().equals(request.visitId()))
```
Comment (zepfred) [inline use-cases/vehicle-routing/src/main/java/org/acme/vehiclerouting/rest/VehicleRoutePlanResource.java]:
```suggestion
                .filter(v -> v.getId().equals(request.visitId()))
```

**Review by zepfred — APPROVED** [review]
The refactoring looks great!

When testing it locally, everything is working fine. However, I noticed that the image's description needs to be updated: `Generate optimal route plan of a vehicle fleet with limited vehicle capacity and time windows. New visits may be added by clicking anywhere inside the map.`

Also, customer and depot terms are still used in the `timefold-solver` documentation. We should keep it consistent with the new domain models.

**Comment by pieterdeschepper** [comment]
There also resides an example for vehicle-routing in the timefold-solver itself. That one still uses depots. Is the documentation referencing that example or the one in the quickstarts?

**Comment by triceo** [comment]
@pieterdeschepper Since we do not have documentation for the VRP quickstart (yet), I expect any references to VRP in the current documentation to refer to the example. Let's not touch the example.

**Pull request merged**